webrender/src/device/gfx/device.rs

```diff
@@ -3254,36 +3302,39 @@ impl<B: hal::Backend> Device<B> {
             self.device.wait_idle().unwrap();
             return;
         }
+        if !self.frame.submitted {
+            return;
+        }
         unsafe {
-            match self.swap_chain.as_mut() {
-                Some(swap_chain) => {
-                    match swap_chain.acquire_image(
-                        !0,
-                        Some(&mut self.image_available_semaphore),
-                        None,
-                    ) {
-                        Ok((id, _)) => {
-                            self.current_frame_id = id as _;
-                            let image = &self.frames[self.current_frame_id].image;
-                            let depth_image = &self.frames[self.current_frame_id].depth.core;
-                            let transitions = image.transit(
+            match self.surface.as_mut() {
+                Some(surface) => {
+                    match surface.acquire_image(!0) {
+                        Ok((surface_image, _)) => {
+                            let depth_image = &self.frame_depths[self.next_id].core;
+                            let new_frame = Frame::new(
+                                self.device.as_ref(),
+                                surface_image,
+                                &depth_image.view,
+                                self.viewport.rect,
+                                &self.render_passes,
+                            );
+                            let old_frame = mem::replace(&mut self.frame, new_frame);
```

Comment:
should we just do `self.frame.surface_image = Some(surface_image))` instead of creating a new one and doing the `mem::replace`?